<launch>
  <!-- generic options -->
  <arg name="vis" default="true" />
  <arg name="camera_enable" default="true"/>
  <arg name="camera_frame_id" default="lighthouse"/>
  <arg name="frame_id" default="odom_meas" /> <!-- frame that we want to report robot pose in -->
  <arg name="count" default="300" /> <!-- how many samples for calibration -->

  <!-- start vive node -->
  <node if="$(arg camera_enable)" name="vive_driver" pkg="vive_tracker_loc"
	type="vive_pose_node" output="screen"/>

  <!-- start calibration node -->
  <node name="system_calibrator" pkg="overhead_mobile_tracker" type="system_calibrator.py"
        output="screen" launch-prefix="xterm -e" respawn="true">
    <param name="frame_id" value="$(arg frame_id)" />
    <param name="camera_frame_id" value="$(arg camera_frame_id)" />
    <param name="count" value="$(arg count)" />
  </node>



  <!-- start rviz -->
  <node if="$(arg vis)" name="rviz" pkg="rviz" type="rviz"
        args="-d $(find overhead_mobile_tracker)/launch/calibration_config.rviz" />

</launch>
